src/hubverse_infrastructure/shared/hubverse_transforms.py

```diff
@@ -0,0 +1,225 @@
+                    "s3:ListBucket",
+                ],
+                principals=[
+                    aws.iam.GetPolicyDocumentStatementPrincipalArgs(
+                        type="Service", identifiers=["lambda.amazonaws.com"]
+                    ),
+                ],
+                resources=[f"arn:aws:s3:::{bucket_name}"],
+            ),
+        ]
+    )
+
+    # Apply the policy to the bucket.
+    aws.s3.BucketPolicy(
+        resource_name=f"{bucket_name}-read-bucket-policy",
+        bucket=hubverse_asset_bucket.id,
+        policy=s3_policy_document.json,
+        opts=ResourceOptions(depends_on=[hubverse_asset_bucket]),
+    )
+
+    return hubverse_asset_bucket
+
+
+def create_cloudwatch_write_policy(policy_name: str) -> aws.iam.Policy:
+    """
+    Create a policy that allows write access to AWS CloudWatch logs
+    (we'll need to attach this to the Lambda execution role)
+    """
+
+    cloudwatch_write_policy_document = aws.iam.get_policy_document(
+        statements=[
+            aws.iam.GetPolicyDocumentStatementArgs(
+                actions=[
+                    "logs:PutLogEvents",
+                    "logs:CreateLogGroup",
+                    "logs:CreateLogStream",
+                ],
+                resources=["arn:aws:logs:*:*:*"],
+            ),
+        ]
+    )
+
+    cloudwatch_write_policy_name = policy_name
+    cloudwatch_write_policy = aws.iam.Policy(
+        name=cloudwatch_write_policy_name,
+        resource_name=cloudwatch_write_policy_name,
+        description="Policy that allows writing to AWS CloudWatch logs",
+        policy=cloudwatch_write_policy_document.json,
+        tags={"hub": "hubverse"},
+    )
+
+    return cloudwatch_write_policy
+
+
+def create_lambda_execution_permissions(lambda_name: str) -> aws.iam.Role:
+    """
+    Create IAM role that the hubverse-transform lambda will assume and attach the necessary permissions.
+    """
+
+    # Because getting ARNs from Pulumi resources is terrible, the code below manually constructs the ARN
+    # of the hubverse-transform lambda function. To do that, we need the current AWS account id and its
+    # default region.
+    aws_account = aws.get_caller_identity().account_id
+    aws_region = aws.get_region().name
+
+    # Create the role used by the lambda function, and limit its use to hubverse-transform lambda function
+    lambda_role_document = aws.iam.get_policy_document(
+        statements=[
+            aws.iam.GetPolicyDocumentStatementArgs(
+                effect="Allow",
+                principals=[
+                    aws.iam.GetPolicyDocumentStatementPrincipalArgs(
+                        type="Service",
+                        identifiers=["lambda.amazonaws.com"],
+                    )
+                ],
+                actions=["sts:AssumeRole"],
+                conditions=[
+                    aws.iam.GetPolicyDocumentStatementConditionArgs(
+                        test="StringEquals",
```

Comment:
Second condition for assuming this role: the request must originate from a specific lambda (i.e., the hubverse transform lambda that we're creating here)